cmd/rpcdaemon/commands/eth_api.go

```diff
@@ -322,32 +322,36 @@ func (api *BaseAPI) getEffectiveGasPricePercentage(tx kv.Tx, txHash common.Hash)
 // APIImpl is implementation of the EthAPI interface based on remote Db access
 type APIImpl struct {
 	*BaseAPI
-	ethBackend      rpchelper.ApiBackend
-	txPool          txpool.TxpoolClient
-	mining          txpool.MiningClient
-	gasCache        *GasPriceCache
-	db              kv.RoDB
-	GasCap          uint64
-	ReturnDataLimit int
-	ZkRpcUrl        string
+	ethBackend                 rpchelper.ApiBackend
+	txPool                     txpool.TxpoolClient
+	mining                     txpool.MiningClient
+	gasCache                   *GasPriceCache
+	db                         kv.RoDB
+	GasCap                     uint64
+	ReturnDataLimit            int
+	ZkRpcUrl                   string
+	AllowFreeTransactions      bool
+	AllowPreEIP155Transactions bool
 }
 
 // NewEthAPI returns APIImpl instance
-func NewEthAPI(base *BaseAPI, db kv.RoDB, eth rpchelper.ApiBackend, txPool txpool.TxpoolClient, mining txpool.MiningClient, gascap uint64, returnDataLimit int, zkRpcUrl string) *APIImpl {
+func NewEthAPI(base *BaseAPI, db kv.RoDB, eth rpchelper.ApiBackend, txPool txpool.TxpoolClient, mining txpool.MiningClient, gascap uint64, returnDataLimit int, zkRpcUrl string, allowFreeTransactions, allowPreEIP155Transactions bool) *APIImpl {
```

Comment:
Technically - yes but I saw that sometimes the zkRpcUrl is empty while allowFreeTransactions, allowPreEIP155Transactions are not. If we replace the three params with a single struct then we will not be able to achieve this. You can check this in AuthAPIList function (daemon.go). Tbh I have no idea why zkRpcUrl was to "" there.

Reply:
Ah ok, right now we're aware that ZK will only work correctly when running erigon as a single unit.  It has the ability to break off parts, such as rpcdaemon.  Currently these won't work in a ZK context so we just pass in "" as the URL because the application just won't work like that.  Something we need to address for sure longer term, but right now you can pass dummy stuff there as it won't work anyway, the focus is on wherever we're passing a "real" value

Reply:
I see. So we're going to use the 3 params rather than a struct for now, right?

Reply:
For me the struct is a nicer approach.  The other places that aren't being used right now (like rpcdaemon) can just take the default to make the compiler happy, we can address all of those in one go later.

Reply:
done